mongoengine/document.py

```diff
@@ -524,12 +527,13 @@ def _save_update(self, doc, save_condition, write_concern):
 
         update_doc = self._get_update_doc()
         if update_doc:
-            upsert = save_condition is None
+            if upsert is None:
+                upsert = save_condition is None
```

Comment:
`save_condition=something` and `upsert=True` are contradictory, perhaps we should raise an exception whenever it enters that condition. If we don't do this, the alternative is to mention which one will prevail on the docstring (and in my opinion save_condition should prevail).

Reply:
Agree, I am going to add a guard at the top asserting that upsert = True and save_condition cannot be set